sound/soc/sof/ipc4-pcm.c

```diff
@@ -429,6 +446,17 @@ static int sof_ipc4_pcm_setup(struct snd_sof_dev *sdev, struct snd_sof_pcm *spcm
 			sof_ipc4_pcm_free(sdev, spcm);
 			return -ENOMEM;
 		}
+
+		if (!support_info)
+			continue;
+
+		stream_info = kzalloc(sizeof(*stream_info), GFP_KERNEL);
+		if (!stream_info) {
+			sof_ipc4_pcm_free(sdev, spcm);
+			return -ENOMEM;
```

Comment:
I am super confused here, sof_ipc4_pcm_free deals with the pipeline_list->pipelines.

Where do we free the stream_info allocated in previous iterations if we take this error branch?

Memory leak?

Reply:
no memory leak. it is free in in sof_ipc4_pcm_free
````c
static void sof_ipc4_pcm_free(struct snd_sof_dev *sdev, struct snd_sof_pcm *spcm)
{
	struct snd_sof_pcm_stream_pipeline_list *pipeline_list;
	int stream;

	for_each_pcm_streams(stream) {
		pipeline_list = &spcm->stream[stream].pipeline_list;
		kfree(pipeline_list->pipelines);
		pipeline_list->pipelines = NULL;
		kfree(spcm->stream[stream].private); // free stream_info
		spcm->stream[stream].private = NULL;
	}
}
````